src/providers/retryProvider.ts

```diff
@@ -260,12 +260,22 @@ export class RetryProvider extends ethers.providers.StaticJsonRpcProvider {
     return response;
   }
 
-  _trySend(provider: ethers.providers.StaticJsonRpcProvider, method: string, params: Array<unknown>): Promise<unknown> {
-    let promise = this._sendAndValidate(provider, method, params);
-    for (let i = 0; i < this.retries; i++) {
-      promise = promise.catch(() => delay(this.delay).then(() => this._sendAndValidate(provider, method, params)));
+  async _trySend(
+    provider: ethers.providers.StaticJsonRpcProvider,
+    method: string,
+    params: Array<unknown>
+  ): Promise<unknown> {
+    let error: unknown;
+    for (let i = 0; i < this.retries; ++i) {
+      try {
```

Comment:
My best attempt at this loop with 0 promise chaining and the clearest control flow would be:
```ts
const retriesLeft = this.retries;
while(true) {
  const [settled] = await Promise.allSettled([this.sendAndValidate(provider, method, params)]);
  if (settled.status === "fulfilled") {
    return settled.value;
  }
  if (retriesLeft-- <= 0) {
    throw settled.reason;
  }
  await delay(this.delay);
}

Reply:
Implemented: https://github.com/across-protocol/sdk/pull/761/commits/bbaa33ac1ba3b9d8c2d22560a3d382e2fba67308